gcc/testsuite/rust/compile/issue-3099.rs

```diff
@@ -0,0 +1,16 @@
+extern "C" {
+    fn printf(s: *const i8, ...);
+}
+
+pub fn main() {
+    unsafe {
+        let value = 10;
+        let result = match value {
+            10 => 15,
+            _ => 20,
+        };
+
+        let format = "Result: %d\n\0" as *const str as *const i8;
```

Comment:
hi @sriganeshres, make sure that you also update your dejagnu regex to assert that you'll get out 15 from println!. 

Without asserting the printf, you're just guarantee that we at least parse it successfully